pocs/utils/images/__init__.py

```diff
@@ -224,29 +232,48 @@ def create_timelapse(directory, fn_out=None, file_type='jpg', **kwargs):
         field_name = head.split('/')[-2]
         cam_name = head.split('/')[-1]
         fname = '{}_{}_{}.mp4'.format(field_name, cam_name, tail)
-        fn_out = os.path.join(os.getenv('PANDIR'), 'images', 'timelapse', fname)
+        fn_out = os.path.normpath(os.path.join(directory, fname))
 
-    fn_dir = os.path.dirname(fn_out)
-    os.makedirs(fn_dir, exist_ok=True)
+    if verbose:
+        print("Timelapse file: {}".format(fn_out))
+
+    if os.path.exists(fn_out) and not overwrite:
+        raise FileExistsError("Timelapse exists. Set overwrite=True if needed")
+
+    ffmpeg = shutil.which('ffmpeg')
     inputs_glob = os.path.join(directory, '*.{}'.format(file_type))
 
     try:
-        ff = FFmpeg(
-            global_options='-r 3 -pattern_type glob',
-            inputs={inputs_glob: None},
-            outputs={
-                fn_out: '-s hd1080 -vcodec libx264'
-            })
-
-        if 'verbose' in kwargs:
-            out = None
-            err = None
-            print("Timelapse command: ", ff.cmd)
-        else:
-            out = open(os.devnull, 'w')
-            err = open(os.devnull, 'w')
-
-        ff.run(stdout=out, stderr=err)
+        ffmpeg_cmd = [
+            ffmpeg,
+            '-r', '3',
+            '-pattern_type', 'glob',
+            '-i', inputs_glob,
+            '-s', 'hd1080',
+            '-vcodec', 'libx264',
+        ]
+
+        if overwrite:
+            ffmpeg_cmd.append('-y')
+
+        ffmpeg_cmd.append(fn_out)
+
+        if verbose:
+            print(ffmpeg_cmd)
+
+        proc = subprocess.Popen(ffmpeg_cmd, universal_newlines=True,
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        try:
+            # Don't wait forever
+            outs, errs = proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            outs, errs = proc.communicate()
+        finally:
+            if verbose:
+                print(outs)
+                print(errs)
+            fn_out = None
```

Comment:
I suspect this belongs in the except block above, along with the `proc.kill()`. Otherwise fn_out is always None when returning.